File: app/src/main/java/com/handstandsam/maintainableespresso/models/ItemBuilder.java
package com.handstandsam.maintainableespresso.models;


public class ItemBuilder {

    Item item;

    public ItemBuilder() {
        item = new Item();
    }

    public ItemBuilder label(String label) {
        item.label = label;
        return this;
    }

    public Item build() {
        return item;
    }
}

File: models/src/main/java/com/handstandsam/shoppingapp/models/builders/ItemBuilder.kt
package com.handstandsam.shoppingapp.models.builders

import com.handstandsam.shoppingapp.models.Item


class ItemBuilder(label: String) {

    internal var item: Item = Item(label, null, null)

    fun label(label: String): ItemBuilder {
        item.label = label
        return this
    }

    fun image(image: String): ItemBuilder {
        item.image = image
        return this
    }

    fun link(link: String): ItemBuilder {
        item.link = link
        return this
    }

    fun build(): Item {
        return item
    }

}
